pkg/state/commitments.go

```diff
@@ -110,3 +110,24 @@ func (c *commitments) size(periodStart uint32) (int, error) {
 	}
 	return len(rec.Commitments), nil
 }
+
+// size returns the number of commitments for the given period start.
+func (c *commitments) generators(periodStart uint32) ([]crypto.PublicKey, error) {
+	key := commitmentKey{periodStart: periodStart}
+	data, err := c.hs.newestTopEntryData(key.bytes())
+	if err != nil {
+		if errors.Is(err, keyvalue.ErrNotFound) {
```

Comment:
If you're looking for data in history use `isNotFoundInHistoryOrDBErr` helper function.